common/model-views.cpp

```diff
@@ -3853,40 +3853,43 @@ namespace rs2
         // it). Lacking an available version, we try to let the user choose a "recommended"
         // version for download. The recommended version is defined by the device (and comes
         // from a #define).
-        if( dev.supports( RS2_CAMERA_INFO_FIRMWARE_VERSION )
-            && dev.supports( RS2_CAMERA_INFO_RECOMMENDED_FIRMWARE_VERSION )
-            && dev.supports( RS2_CAMERA_INFO_PRODUCT_LINE ) )
+
+        // 'notification_type_is_displayed()' is used to detect if fw_update notification is on to avoid displaying it during FW update process when
+        // the device enters recovery mode
+        if( ! not_model->notification_type_is_displayed< fw_update_notification_model >()
+            && ( dev.is< updatable >() || dev.is< update_device >() ) )
         {
-            std::string fw = dev.get_info( RS2_CAMERA_INFO_FIRMWARE_VERSION );
-            std::string recommended_fw_ver
-                = dev.get_info( RS2_CAMERA_INFO_RECOMMENDED_FIRMWARE_VERSION );
+            std::string fw;
+            std::string recommended_fw_ver;
+            int product_line = 0;
+
+            // Override with device info if info is available
+            if (dev.is<updatable>())
+            {
+                fw = dev.get_info(RS2_CAMERA_INFO_FIRMWARE_VERSION);
```

Comment:
You haven't checked if it's supported... I realize there's a slim chance but still.